File: Python/ps_baekjoon/11724/main.py
# baekjoon ps 11724

import sys
import queue

n, m = map(int, input().split(" "))
cnt = 0
data = [[] for i in range(n + 1)]
check = [False for i in range(n + 1)]
for _ in range(m):
    a, b = map(int, input().split(" "))
    data[a].append(b)
    data[b].append(a)
q = queue.Queue()
for i in range(1,n+1):
    if not check[i]:
        q.put(i)
        cnt+=1
        check[i] = True
    while not q.empty():
        for i in data[q.get()]:
            if not check[i]:
                q.put(i)
                check[i] = True
print(cnt)

File: C++/ps_baekjoon/11724/main.cpp
//baekjoon ps 11724

#include <iostream>
#include <queue>
#include <vector>

using namespace std;

int main()
{
    ios_base::sync_with_stdio(false);
    int n, m;
    int cnt = 0;
    cin >> n >> m;
    vector<vector<int>> map(n + 1);
    vector<bool> check(n + 1, false);
    queue<int> q;
    while (m--)
    {
        int a, b;
        cin >> a >> b;
        map[a].push_back(b);
        map[b].push_back(a);
    }
    for (int i = 1; i <= n; i++)
    {
        if (!check[i])
        {
            q.push(i);
            cnt++;
            check[i] = true;
        }
        while (!q.empty())
        {
            int fn = q.front();
            q.pop();
            for (const auto e : map[fn])
                if (!check[e])
                {
                    q.push(e);
                    check[e] = true;
                }
        }
    }
    cout << cnt;
    return 0;
}